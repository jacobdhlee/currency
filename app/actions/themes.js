export const PRIMARY_COLOR_CHANGE = 'PRIMARY_COLOR_CHANGE';

export const changeTheme = color => ({
  type: PRIMARY_COLOR_CHANGE,
  color,
});
