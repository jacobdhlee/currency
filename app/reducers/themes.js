import { PRIMARY_COLOR_CHANGE } from '../actions/themes';

const INITIAL_STATE = {
  primaryColor: '#4F6D7A',
};

const themeReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case PRIMARY_COLOR_CHANGE:
      return {
        ...state,
        primaryColor: action.color,
      };
    default:
      return state;
  }
};

export default themeReducer;
