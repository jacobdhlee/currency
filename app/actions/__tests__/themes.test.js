import { changeTheme } from '../themes';

it('change PRIMARY_COLOR_CHANGE test', () => {
  expect(changeTheme('red')).toMatchSnapshot();
});
