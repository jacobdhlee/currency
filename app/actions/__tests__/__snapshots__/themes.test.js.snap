// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change PRIMARY_COLOR_CHANGE test 1`] = `
Object {
  "color": "red",
  "type": "PRIMARY_COLOR_CHANGE",
}
`;
